src/test/scala/tech/cryptonomic/conseil/tezos/MetadataCachingTest.scala

```diff
@@ -0,0 +1,115 @@
+package tech.cryptonomic.conseil.tezos
+
+import cats.effect._
+import cats.effect.concurrent.MVar
+import com.rklaehn.radixtree.RadixTree
+import org.scalatest.{Matchers, OneInstancePerTest, WordSpec}
+import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.{Attribute, DataType, Entity, KeyType}
+import tech.cryptonomic.conseil.tezos.MetadataCaching._
+
+import scala.concurrent.ExecutionContext
+
+class MetadataCachingTest extends WordSpec with Matchers with OneInstancePerTest {
+
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+
+
+  val cachingStatus: MVar[IO, CachingStatus] = MVar[IO].of[CachingStatus](NotStarted).unsafeRunSync()
+  val attributesCache: MVar[IO, AttributesCache] = MVar[IO].empty[AttributesCache].unsafeRunSync()
+  val entitiesCache: MVar[IO, EntitiesCache] = MVar[IO].empty[EntitiesCache].unsafeRunSync()
+  val attributeValuesCache: MVar[IO, AttributeValuesCache] = MVar[IO].empty[AttributeValuesCache].unsafeRunSync()
+
+  val sut: MetadataCaching[IO] = new MetadataCaching[IO](cachingStatus, attributesCache, entitiesCache, attributeValuesCache)
+
+  "Metadata caching" should {
+    "get initial state of caching status" in {
+      sut.getCachingStatus.unsafeRunSync() shouldBe NotStarted
+    }
+
+    "update cache status" in {
+      sut.updateCachingStatus(InProgress).unsafeRunSync()
```

Comment:
I'd suggest `sut` to be rebuilt from scratch for each test, otherwise the test cases will be entangled

Reply:
I've mixed `with OneInstancePerTest` and it's being handled

Reply:
awesome, didn't knew about that